shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedProcessorSyncServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.redhat.service.smartevents.shard.operator.v2;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.infra.v2.api.models.ConditionStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ConditionDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorStatusDTO;
+import com.redhat.service.smartevents.shard.operator.v2.converters.ProcessorStatusConverter;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedProcessor;
+
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_PROCESSOR_DELETED_NAME;
+
+@ApplicationScoped
+public class ManagedProcessorSyncServiceImpl implements ManagedProcessorSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorSyncServiceImpl.class);
+
+    @Inject
+    ManagerClient managerClient;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Override
+    public void syncManagedProcessorWithManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .invoke(this::processDelta)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully processed ManagedProcessor deltas"),
+                        error -> LOGGER.debug("Failed to process ManagedProcessor deltas"));
+    }
+
+    @Override
+    public void syncManagedProcessorStatusBackToManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .transform(this::transformToProcessorStatus)
+                .invoke(this::notifyProcessorStatus)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully sync ManagedProcessor status with Manager"),
+                        error -> LOGGER.error("Failed to sync ManagedProcessor status with Manager", error));
+    }
+
+    private void processDelta(List<ProcessorDTO> processorDTOList) {
+        for (ProcessorDTO processorDTO : processorDTOList) {
```

Comment:
Something @r00ta pointed out to me on one of my PRs that may be desirable here too.

If processing of any of the items in the `List` results in an exception being thrown the remainder of the `List` will be skipped. It might be better defensive programming to catch and log unexpected exceptions within the loop to ensure other items can be processed? The same can be said for other places where you iterate loops of resources.

Reply:
this is the not the best usecase to implement defensive programming.